testing/configure/azure_scripts/lib/utilityFunctions.ps1

```diff
@@ -0,0 +1,143 @@
+function Format-AzVmRunCommandOutput {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$JsonResponse
+    )
+
+    $results = @()
+
+    try {
+        $responseObj = $JsonResponse | ConvertFrom-Json
+#        Write-Output "Converted JSON object: $responseObj"
+
+        if ($responseObj -and $responseObj.value) {
+            $stdout = ""
+            $stderr = ""
+
+            foreach ($item in $responseObj.value) {
+#                Write-Output "Processing item: $($item.code)"
+
+                # Check for StdOut and StdErr
+                if ($item.code -like "ComponentStatus/StdOut/*") {
+                    $stdout += $item.message + "`n"
+                } elseif ($item.code -like "ComponentStatus/StdErr/*") {
+                    $stderr += $item.message + "`n"
+                }
+
+                # Additional case to handle other types of 'code'
+                # This ensures that all messages are captured
+                else {
+                    $stdout += $item.message + "`n"
+                }
+            }
+
+            if ($stdout -or $stderr) {
+                $results += New-Object PSObject -Property @{
+                    StdOut = $stdout
+                    StdErr = $stderr
+                }
+            }
+        }
+    } catch {
+        $errorMessage = $_.Exception.Message
+        Write-Output "Error: $errorMessage"
+        $results += New-Object PSObject -Property @{
+            StdOut = "Error: $errorMessage"
+            StdErr = ""
+        }
+    }
+
+    if (-not $results) {
+        $results += New-Object PSObject -Property @{
+            StdOut = "No data or invalid data received."
+            StdErr = ""
+        }
+    }
+
+    return $results
+}
+
+function Show-FormattedOutput {
+    param (
+        [Parameter(Mandatory = $true)]
+        [Object[]]$FormattedOutput
+    )
+
+    foreach ($item in $FormattedOutput) {
+        if ($item -is [string]) {
+            # Handle string messages (like error or informational messages)
+            Write-Output $item
+        }
+        elseif ($item -is [PSCustomObject]) {
+            # Handle custom objects with StdOut and StdErr
+            if (![string]::IsNullOrWhiteSpace($item.StdOut)) {
+                Write-Output "Output (stdout):"
+                Write-Output $item.StdOut
+            }
+            if (![string]::IsNullOrWhiteSpace($item.StdErr)) {
+                Write-Output "Error (stderr):"
+                Write-Output $item.StdErr
+            }
+        }
+    }
+}
+
+function ExtractPrivateKeyFromJson {
```

Comment:
Might want to rename this function to match PowerShell's recommended approved verb syntax. https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.4

Maybe function Get-PrivateKeyFromJson?

Reply:
Done